### ISSUE TITLE
Redundant threadIndeX and threadIndexY as arguments of device functions.

### ISSUE THREAD

**Issue by Mamatu** [Mamatu/oap#7]
Some functions pass as argument two integers variables: threadIndexX and threadIndexY - which are indecies of threads in grid. It should be removed and this data should be acquired directly from cuda statements in scope of function
```
__device__ func() {
  uintt threadIndexX = blockIdx.x * blockDim.x + threadIdx.x;
  uintt threadIndexY = blockIdx.y * blockDim.y + threadIdx.y;
}  
```

**Comment by Mamatu** [comment]
Created branch for this issue : issues/7. Should be checked with device code.

**Comment by Mamatu** [comment]
Branch issues/7 was merged in squashed commit: commit 6921502d0889f11a1481054e7c5488b1325e92f3
